src/nvim/message.c

```diff
@@ -241,12 +240,12 @@ msg_strtrunc (
  * Truncate a string "s" to "buf" with cell width "room".
  * "s" and "buf" may be equal.
  */
-void trunc_string(char_u *s, char_u *buf, int room, int buflen)
+void trunc_string(char_u *s, char_u *buf, long room, long buflen)
 {
-  int half;
-  int len;
+  long half;
+  long len;
+  long i;
```

Comment:
I think `size_t` would work better. No casts needed after `STRLEN()`, no comparisons with unsigned values, and we already assume `>= 0` for each.